toplevel/ccompile.ml

```diff
@@ -147,19 +149,22 @@ let compile opts stm_options injections copts ~echo ~f_in ~f_out =
       let _doc = Stm.join ~doc:state.doc in
       let wall_clock2 = Unix.gettimeofday () in
       check_pending_proofs long_f_dot_in;
-      (* In .vo production, dump a complete .vo file.
-         In .vok production, only dump an empty .vok file. *)
+      (* In .vo production, dump a complete .vo file. *)
       if mode = BuildVo
         then Library.save_library_to ~output_native_objects Library.ProofsTodoNone ldir long_f_dot_out;
       Aux_file.record_in_aux_at "vo_compile_time"
         (Printf.sprintf "%.3f" (wall_clock2 -. wall_clock1));
       Aux_file.stop_aux_file ();
-      (* In .vo production, dump an empty .vos file to indicate that the .vo should be loaded,
-         and dump an empty .vok file to indicate that proofs are ok. *)
+      (* Additionally, dump empty .vos and .vok files to make sure that
+        stale ones are never loaded *)
       if mode = BuildVo then begin
         dump_empty_vos();
-        create_empty_file (long_f_dot_out ^ "k");
+        dump_empty_vok();
       end;
+      (* In .vok production mode, dump an empty .vok file to
+         indicate that proofs are ok. *)
+      if mode = BuildVok then
+        dump_empty_vok();
```

Comment:
Here we're in either `BuildVo` or `BuildVok` mode, so either way at this point we should `dump_empty_vok()`.